### ISSUE TITLE
p2 subdomain won't load on production

### ISSUE THREAD

**Issue by gnodar01** [susanBuck/dwa15-fall2017#34]
The notes say that if a change in the domain's DNS hasn't occurred after an hour or so, there might be an error. When setting up my p2 subdomain, I followed the same steps as p1, but it still doesn't load. It's been several hours. I suppose it may still be propagating, but I don't want to risk scrambling on a Tuesday after realizing there was an error in my config somewhere. My p1 subdomain is still working just fine.

Does anyone spot anything that may be misconfigured?

### Config

```
root@cscie15:~# cd /var/www/html/
root@cscie15:/var/www/html# ls
CSCIE15-p1  CSCIE15-p2  index.html  info.php
root@cscie15:/var/www/html# cd CSCIE15-p2
root@cscie15:/var/www/html/CSCIE15-p2# ls
index.php  README.md
root@cscie15:/var/www/html/CSCIE15-p2# cat index.php
<!DOCTYPE html>
<html>
<head>

	<title>Hello World</title>
	<meta charset='utf-8'>

</head>
<body>

	<?php
	echo 'Hello World!'
	?>
</body>
</html>
root@cscie15:/var/www/html/CSCIE15-p2# cat /etc/apache2/sites-enabled/000-default.conf
<VirtualHost *:80>
        ServerAdmin webmaster@localhost
        DocumentRoot /var/www/html

        <Directory /var/www/html/>
            Options Indexes FollowSymLinks
            AllowOverride All
            Require all granted
        </Directory>

        ErrorLog ${APACHE_LOG_DIR}/error.log
        CustomLog ${APACHE_LOG_DIR}/access.log combined

        <IfModule mod_dir.c>
            DirectoryIndex index.php index.pl index.cgi index.html index.xhtml index.htm
        </IfModule>

</VirtualHost>

<VirtualHost *:80>
  ServerName p1.nodar.me
  DocumentRoot "/var/www/html/CSCIE15-p1"
  <Directory "/var/www/html/CSCIE15-p1">
    AllowOverride All
  </Directory>
</VirtualHost>

<VirtualHost *:80>
  ServerName p2.nodar.me
  DocumentRoot "/var/www/html/CSCIE15-p2"
  <Directory "/var/www/html/CSCIE15-p2">
    AllowOverride All
  </Directory>
</VirtualHost>
root@cscie15:/var/www/html/CSCIE15-p2# service apache2 restart
```

### Error log

```
root@cscie15:/var/log/apache2# cat error.log
[Sun Sep 17 06:25:03.011000 2017] [mpm_prefork:notice] [pid 14426] AH00163: Apache/2.4.18 (Ubuntu) configured -- resuming normal operations
[Sun Sep 17 06:25:03.011118 2017] [core:notice] [pid 14426] AH00094: Command line: '/usr/sbin/apache2'
[Mon Sep 18 01:58:30.759121 2017] [mpm_prefork:notice] [pid 14426] AH00169: caught SIGTERM, shutting down
[Mon Sep 18 01:58:31.892802 2017] [mpm_prefork:notice] [pid 6853] AH00163: Apache/2.4.18 (Ubuntu) configured -- resuming normal operations
[Mon Sep 18 01:58:31.892908 2017] [core:notice] [pid 6853] AH00094: Command line: '/usr/sbin/apache2'
[Mon Sep 18 02:52:18.288837 2017] [mpm_prefork:notice] [pid 6853] AH00169: caught SIGTERM, shutting down
[Mon Sep 18 02:52:19.678875 2017] [mpm_prefork:notice] [pid 7113] AH00163: Apache/2.4.18 (Ubuntu) configured -- resuming normal operations
[Mon Sep 18 02:52:19.679058 2017] [core:notice] [pid 7113] AH00094: Command line: '/usr/sbin/apache2'
[Mon Sep 18 03:08:09.875997 2017] [mpm_prefork:notice] [pid 7113] AH00169: caught SIGTERM, shutting down
[Mon Sep 18 03:08:11.047012 2017] [mpm_prefork:notice] [pid 7494] AH00163: Apache/2.4.18 (Ubuntu) configured -- resuming normal operations
[Mon Sep 18 03:08:11.047168 2017] [core:notice] [pid 7494] AH00094: Command line: '/usr/sbin/apache2'
[Mon Sep 18 03:29:13.226518 2017] [mpm_prefork:notice] [pid 7494] AH00169: caught SIGTERM, shutting down
[Mon Sep 18 03:29:14.358744 2017] [mpm_prefork:notice] [pid 7707] AH00163: Apache/2.4.18 (Ubuntu) configured -- resuming normal operations
[Mon Sep 18 03:29:14.358871 2017] [core:notice] [pid 7707] AH00094: Command line: '/usr/sbin/apache2'
```

### DNS settings
![screen shot 2017-09-17 at 11 41 00 pm](https://user-images.githubusercontent.com/8888732/30528580-68bedfbc-9c02-11e7-95f5-e46cbb36ffb2.png)

**Comment by gnodar01** [comment]
Worth noting that everything works on local, my repos are synched, and I've tried on multiple devices so it shouldn't be a caching issue.

**Comment by ysrinu** [comment]
@gnodar01 , your cscie15-p2 folder looks fine as evident from http://nodar.me/CSCIE15-p2/
Can you resave the p2 (namecheap) record and restart apache (digital ocean)?

**Comment by gnodar01** [comment]
@ysrinu Did both and (after 30 minutes) hasn't worked.

**Comment by ysrinu** [comment]
@gnodar01 , I could ping your p1 but not p2. I would like to think this means the subdomain is not pointed to your linux server IP address yet, and that this is irrespective of whether your apache virutal host is configured for this subdomain or not.

![image](https://user-images.githubusercontent.com/25366909/30530334-e922c04a-9c13-11e7-9ea1-7ccc3e6074dc.png)

Can you look into options 3 and 4 of below:
https://www.namecheap.com/support/knowledgebase/article.aspx/9622/10/dns-propagation--explained

**Comment by gnodar01** [comment]
@ysrinu Thanks for the link. I changed TTL from automatic to 5 minutes, and without continuing the rest of the steps outlined in option 3 it showed up immediately. Unless it just happened to go live at the same moment as when I changed that setting, I still don't totally understand what the issue was. It wasn't some local caching issue, and I tried visiting the site with my phone, with wifi off, meaning I tried from separate networks. So I don't think it was simply an ISP caching issue either.

In any case, it's working now. Thanks for your time.
